pilot/cmd/pilot-agent/status/server.go

```diff
@@ -258,13 +271,22 @@ func (s *Server) handleStats(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
+	if istioAgent, err = s.scrape(fmt.Sprintf("http://localhost:%d/agent-health", s.statusPort), r.Header); err != nil {
```

Comment:
we should be able to replace this with `promStatsExporter.ServeHTTP(w, r)`, right?

Reply:
how will I capture the written response? using an `httptest` recorder?